packages/frontend/src/redux/actions/staking.js

```diff
@@ -533,3 +525,61 @@ export const handleUpdateCurrent = (accountId) => async (dispatch, getState) =>
     let currentAccount = selectStakingCurrentAccountbyAccountId(getState(), { accountId });
     dispatch(staking.updateCurrent({ currentAccount }));
 };
+
+export const getValidatorFarmData = (validatorId, accountId) => async (dispatch, getState) => {
+
+    const validators = selectStakingAllValidators(getState());
+    const validator = { ...validators.find(validator => validator?.accountId === validatorId)};
+
+    if (validator?.version !== FARMING_VALIDATOR_VERSION) return;
+
+    const poolSummary = await validator.contract.get_pool_summary();
+    const farms = await validator.contract.get_farms({ from_index: 0, limit: 300 });
+    
+ 
+    const list = await Promise.all(farms.map(({ token_id, farm_id }) => {
+        try {
+            dispatch(fetchToken({ contractName: token_id }));
+            return validator.contract
+                .get_unclaimed_reward({ account_id: accountId, farm_id })
+                .catch(() => "0")
+                .then((balance) => ({ 
+                    token_id,
+                    balance,
+                    farm_id,
+                }));
+        } catch (error) {
+            console.error(error);
```

Comment:
Handling errors, in this case, should be done with modal or smth more suitable.
If any request failed - the promise will be rejected. @MaximusHaximus provides information on how it should be done properly.